web/packages/teleport/src/components/UserMenuNav/UserMenuNav.tsx

```diff
@@ -160,7 +169,35 @@ export function UserMenuNav({ username }: UserMenuNavProps) {
 
   return (
     <Container ref={ref}>
-      <UserInfo onClick={() => setOpen(!open)}>
+      <UserInfo
+        onClick={() => setOpen(!open)}
+        onKeyUp={e => {
+          if (e.key === 'Enter') {
+            setOpen(!open);
+            return;
+          }
+          if (e.key === 'Tab' && open) {
+            // move to first focusable item in dropdown
+            dropdownRef.current
+              ?.querySelector<HTMLElement>('a, div[role="button"]')
+              ?.focus();
+          }
+        }}
+        onBlur={e => {
+          if (
+            !e.currentTarget.contains(e.relatedTarget as Node) &&
+            !dropdownRef.current?.contains(e.relatedTarget as Node)
```

Comment:
I really like this behavior. However, could you either comment or move individual checks to named variables? It's clear for me at the moment what they're for, but in a few months it'll be harder to skim through this.

Now that I think about it, even capturing it in a function called something like `closeOnFocusOutside` would be helpful. Though the intent behind `!e.currentTarget.contains(e.relatedTarget as Node)` would still not be easy to understand without looking up the docs for `e.relatedTarget`.

Reply:
For sure, can try to clarify this better

Reply:
Extracted to a more generic func as I think there are more places this would be useful:
https://github.com/gravitational/teleport/blob/23a10fa0d6f2396fc6d478378c69403030475bae/web/packages/teleport/src/lib/util.ts#L26-L31